bigquery/google/cloud/bigquery/_helpers.py

```diff
@@ -328,10 +328,15 @@ def __init__(self, udf_type, value):
         self.value = value
 
     def __eq__(self, other):
+        if not isinstance(other, UDFResource):
+            return NotImplemented
         return(
             self.udf_type == other.udf_type and
             self.value == other.value)
 
+    def __ne__(self, other):
+        return not self == other
```

Comment:
I am currently poring over an "examination" of how these calls works: https://gist.github.com/dhermes/3e3ad5b6aceabaacc3898fedbe040ff9

I don't fully understand it, so I'm trying.